front/lib/actions/process.ts

```diff
@@ -162,30 +165,40 @@ export class ProcessActionType extends BaseAction {
 
     content += "PROCESSED OUTPUTS:\n";
 
-    if (this.outputs) {
-      if (this.outputs.data.length === 0) {
-        content += "(none)\n";
-      } else {
-        for (const o of this.outputs.data) {
-          content += `${JSON.stringify(o)}\n`;
-        }
+    if (!this.outputs) {
+      return {
```

Comment:
Why do we add those emulated function calls when you said we will want to remove them after this refacto is done? 🙏🏻